.github/workflows/codespell.yaml

```diff
@@ -0,0 +1,20 @@
+name: codespell
+on:
+  pull_request:
+    branches:
+      - master
+      - enterprise
+permissions: {}
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+      - uses: codespell-project/actions-codespell@master
+        with:
+          ignore_words_list: "ser"
+          skip: "./.git,build/*,tools/*,*.js,*.thrift"
```

Comment:
I believe we need to ignore Seastar as well (generally, all submodules) in Scylla (we should / can fix them in their original repos?)

Reply:
see https://github.com/actions/checkout/blob/main/README.md#usage and https://github.com/actions/checkout/blob/b4ffde65f46336ab88eb53be808477a3936bae11/action.yml#L82-L90 . 
i am quoting the document here as well:
```yaml
    # Whether to checkout submodules: `true` to checkout submodules or `recursive` to
    # recursively checkout submodules.
    #
    # When the `ssh-key` input is not provided, SSH URLs beginning with
    # `git@github.com:` are converted to HTTPS.
    #
    # Default: false
    submodules: ''
```
by default, `submodules` is false. so the submodules are not fetched.

Reply:
Another directory you should maybe consider excluding is test/cql. These look like text files, but really aren't, and "fixing" words there can break the tests. Although usually, if you fix both the .cql and .result file in the same way, it should be fine.

Reply:
thanks. added.